packages/core/src/interpreter.ts

```diff
@@ -1208,11 +1210,12 @@ export class Interpreter<
       id,
       send: (event) => receivers.forEach((receiver) => receiver(event)),
       subscribe: (next) => {
-        listeners.add(next);
+        const observer = toObserver(next);
+        listeners.add(observer.next);
```

Comment:
this is a fix from the perspective of the `Symbol.observable` - since that method is accepting observers we convert the received callback (or observer) to an observer